cmd/rootMain.go

```diff
@@ -125,24 +125,41 @@ Try 'help'`)
 		return exitCode{1}
 	}
 
-	codeCh := make(chan int)
+	exitCodeCh := make(chan int)
 	go func() {
 		if runErr := comm.Wait(); runErr != nil {
 			if exitErr, ok := runErr.(*exec.ExitError); ok {
-				codeCh <- exitErr.ExitCode()
+				exitCodeCh <- exitErr.ExitCode()
 			} else {
 				cmdutil.SafeErrPrintf("%v\n", runErr)
-				codeCh <- 1
+				exitCodeCh <- 1
 			}
 		} else {
-			codeCh <- 0
+			exitCodeCh <- 0
 		}
-		close(codeCh)
+		close(exitCodeCh)
 	}()
 
+	// Only switch process group if interactive, where the child starts a new session.
```

Comment:
Nit: This should probably contain some details about why we're doing this for future maintainers. Thinking of something like

> Only switch process group if interactive, where the child starts a new session. This way, the daemon and the shell share the same TTY, which enables plugins to prompt the user for input like e.g. security tokens.